### ISSUE TITLE
Error on application start when including imageio-batik

### ISSUE THREAD

**Issue by popov654** [haraldk/TwelveMonkeys#721]
```Could not instantiate SVGImageReader (missing support classes).
java.lang.NoClassDefFoundError: org/apache/batik/transcoder/TranscoderException
        at com.twelvemonkeys.imageio.plugins.svg.SVGImageReaderSpi.onRegistration(SVGImageReaderSpi.java:177)
        at javax.imageio.spi.SubRegistry.registerServiceProvider(ServiceRegistry.java:715)
        at javax.imageio.spi.ServiceRegistry.registerServiceProvider(ServiceRegistry.java:302)
        at javax.imageio.spi.IIORegistry.registerApplicationClasspathSpis(IIORegistry.java:211)
        at javax.imageio.spi.IIORegistry.<init>(IIORegistry.java:138)
        at javax.imageio.spi.IIORegistry.getDefaultInstance(IIORegistry.java:159)
        at javax.imageio.ImageIO.<clinit>(ImageIO.java:65)
        at render.TestWebPImage.<init>(TestWebPImage.java:25)
        at render.TestWebPImage.main(TestWebPImage.java:48)
Caused by: java.lang.ClassNotFoundException: org.apache.batik.transcoder.TranscoderException
        at java.net.URLClassLoader$1.run(URLClassLoader.java:366)
        at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:425)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
```

I tried the library compiled by myself and the provided on the page, they are both giving errors.

I know someone asked here already, but I did not understand how he solved the issue. I guess I need some third party library, that was not included or downloaded by Maven automatically. How can I make it work?

**Comment by popov654** [comment]
Tried to put in Apache Batik from their site, but it throws some error about unexpected class version. I guess that's because it is compiled against Java 8

**Comment by haraldk** [comment]
You could try the same approach, compiling Batik against Java 7 (but I have no idea if that will work). Otherwise you'll need to find an older version of Batik that works with Java 7 (but beware of security issues with older Batik versions). I'm afraid I can't help with third party libraries. 

You might eventually find that upgrading your own application to Java 8 (or preferably later) will be easier... 😀

**Comment by popov654** [comment]
Unfortunately I cannot upgrade, because I'm using Swing Application Framework, that was dropped from version 8 and up.
Tried downloading Batik sources and build it as per instructions. I had to change Java Version in pom.xml from 1.8 to 1.7, but Maven is ignoring this value for some reason, trying to use 1.8. What is going on? :)

Do you know the latest version of Batik compiled for Java 7?

**Comment by haraldk** [comment]
No, sorry. You have to consult their documentation.

For the Maven build, make sure you configure javac (compile) plugin to output 1.7 classes. It's not enough to compile against the 1.7 API, if you want to run on a 1.7 JVM... 

I've used SAF before too, and was a bit sad when Oracle dropped it for the JDK. I'm sure there are forks that work with more updated Java versions, like BSAF etc.

**Comment by popov654** [comment]
It seems like I managed to build for 1.7, I had to correct 1 more file, build.xml.

About SAF - I don't know what is the issue. I mean, older code absolutely must run on newer JRE versions, and SAF is fully written in Java as a wrapper upon Swing (for easy resource management, better and cleaner architecture, and so on). It does not include anything native, that could get broken, it's just plain Java sources. And for some reason newer IDEs like NetBeans 8.x fail to load projects using it. Seems strange for me.

Maybe the Swing changed inside itself with Java 8, and SAF authors just decided to quit maintaining the project to adapt to newer versions of Swing, idk.

But they could still provide a way to use older Swing libraries with new Java.
